cmd/tusd/cli/serve.go

```diff
@@ -92,7 +113,42 @@ func Serve() {
 		stdout.Printf("You can now upload files to: http://%s%s", address, basepath)
 	}
 
-	if err = http.Serve(listener, nil); err != nil {
-		stderr.Fatalf("Unable to serve: %s", err)
+	var g run.Group
```

Comment:
Can you explain what the run.Group is? Do we need it?

Reply:
To be honest, the run.Group usage makes the code a lot harder for me and other contributors to understand. This package might be very helpful for larger application with many moving parts but I don't think it's necessary in this case. There are some very simple and easy to understand examples about stopping a HTTP server from a signal. For example, https://stackoverflow.com/a/40041517. What do you think about switching to that one?

Reply:
I'm not really interested in rewriting the patch. Feel free to rewrite it if you understand how to do so.

However, is there something in particular that you and the other contributors you've discussed this with don't understand about oklog/run groups? Peter Bourgon (the oklog/run author) is also accessible on the gophers slack and may be able to clarify regarding any specific questions you have.